pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
> It looks at all the samples.
Ah, I see. Someone told me it ignores values between steps. So if you query with a large step, values in between steps will be ignored. Is this not true for rate function due to it needing to detect rests? 

For example if you query increase(logins.count[1d]) over last 2 months with step of 60 (or a 100). will that be accurate? won't the step parameter need to be as low as sample rate to detect resets and calculate rate accurately?